implementation/src/main/java/io/smallrye/config/lifecycle/ConfigSourceInterceptorContext.java

```diff
@@ -0,0 +1,35 @@
+package io.smallrye.config.lifecycle;
+
+public interface ConfigSourceInterceptorContext {
+    /**
+     * This will provide the ConfigSourceInterceptor with the lookup value of the key on a ConfigSource.
+     *
+     * Should the ConfigSource be exposed?
+     *
+     * @return a ConfigValue
+     */
+    ConfigValue getConfigValue();
+
+    /**
+     * Lookup a key only into the current or previously evaluated ConfigSources
+     *
+     * @param key the key to lookup.
+     * @return the lookup key value.
+     */
+    String getValue(String key);
+
+    /**
+     * Proceeds to the next interceptor.
+     *
+     * @return the config key lookup value.
+     */
+    String proceed();
+
+    /**
+     * Proceeds to the next interceptor.
+     *
+     * @param value the transformed value.
+     * @return the config key lookup value.
+     */
+    String proceed(String value);
```

Comment:
This would change to:
```java
ConfigValue proceed(String name);
```

Reply:
Ok, in here we need to discuss a little further both or approaches and figure out what we want to do. See my previous comment in the applyInterceptors method review.